### ISSUE TITLE
should inferring gcc-toolchain be limited to bundled LLVM?

### ISSUE THREAD

**Issue by mppf** [chapel-lang/chapel#19518]
Today, `printchplenv` gathers configuration information for the LLVM backend by looking for `which gcc` and then running `gcc -v` and looking for a `--prefix=` argument. It uses the result of that as a `--gcc-toolchain=` argument for `clang`.

Today we do this for CHPL_LLVM=system and for CHPL_LLVM=bundled. However, I am not so sure that we want it to be the default behavior in as many cases as it is today. In particular, if you install a system LLVM and want to use the GCC C++ standard library headers, you should probably have configured the system LLVM with `-DGCC_INSTALL_PREFIX` (see https://clang.llvm.org/get_started.html ).

Different use cases seem to want a different default:
 * On a Cray XC system we know that the version of `gcc` can change as different modules are loaded and we'd like to respond to that.
 * On a linux system with Homebrew, using `gcc` is not desired and in fact the logic to find the gcc toolchain causes problems when Homebrew is using a different gcc than the system default

This issue proposes that we limit this behavior to `CHPL_LLVM=bundled` and potentially make an exception for `CHPL_LLVM=system` on XC.

<details>

https://github.com/chapel-lang/chapel/blob/f9a5589a6b32fbdc7ebb6fe72fe2fff87d19a763/util/chplenv/chpl_llvm.py#L447-L462

https://github.com/chapel-lang/chapel/blob/f9a5589a6b32fbdc7ebb6fe72fe2fff87d19a763/util/chplenv/chpl_llvm.py#L305-L330

</details>

**Comment by bradcray** [comment]
Not being so familiar with this logic:  Do we also look for `gcc` no matter what CHPL_HOST_COMPILER is set to?

**Comment by mppf** [comment]
> Not being so familiar with this logic:  Do we also look for `gcc` no matter what CHPL_HOST_COMPILER is set to?

We do. But it's not so clear to me if having the behavior depend on `CHPL_HOST_COMPILER` is reasonable since this inference is really about how we configure LLVM/clang (so `CHPL_LLVM` and `CHPL_TARGET_COMPILER` seem to be the ones that should obviously be related).

**Comment by bradcray** [comment]
Ah, OK.  I misunderstood this query as being more about building LLVM than how we invoke LLVM.  Thanks.

**Comment by mppf** [comment]
Right, we could use it only when building LLVM, but that would cause problems on XC module builds. (But maybe then XC should be the exception)?

One thing we could do is only infer it when requested to do so.
